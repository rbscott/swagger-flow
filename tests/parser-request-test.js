import yaml from 'js-yaml';

import parse from '../src/parser';

describe('parse', () => {
  describe('parameters', () => {
    it('Should parse a basic parameter', () => {
      const input = yaml.safeLoad(`
paths:
  /simple/test:
    get:
      parameters:
      - name: param1
        in: query
        type: string
`);
      expect(parse(input).items).toMatchSnapshot();
    });

    it('Should parse multiple parameters', () => {
      const input = yaml.safeLoad(`
paths:
  /simple/test:
    get:
      parameters:
      - name: param1
        in: query
        type: string
      - name: param2
        in: query
        type: boolean
`);
      expect(parse(input).items).toMatchSnapshot();
    });

    it('Should parse global parameters', () => {
      const input = yaml.safeLoad(`

parameters:
- name: globalParam1
  in: query
  type: string

paths:
  /simple/test:
    get:
      parameters:
      - name: param1
        in: query
        type: string
`);
      expect(parse(input).items).toMatchSnapshot();
    });

    it('Should parse path parameters', () => {
      const input = yaml.safeLoad(`

paths:
  /simple/test:
    parameters:
    - name: pathParam1
      in: query
      type: string
    get:
      parameters:
      - name: param1
        in: query
        type: string
`);
      expect(parse(input).items).toMatchSnapshot();
    });

    it('Should inherit properly', () => {
      const input = yaml.safeLoad(`

parameters:
- name: sameParam
  in: query
  type: string

paths:
  /simple/test:
    parameters:
    - name: sameParam
      in: query
      type: boolean
    get:
      parameters:
      - name: sameParam
        in: query
        type: number
`);
      expect(parse(input).items).toMatchSnapshot();
    });

  });
});
