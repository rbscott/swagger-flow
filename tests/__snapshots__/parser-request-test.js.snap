// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse parameters Should inherit properly 1`] = `
Array [
  Object {
    "additionalProperties": false,
    "fields": Array [
      Array [
        "sameParam",
        Object {
          "id": "#/simple/test/get/sameParam",
          "name": "RqstSimpleTestGet_Field_sameParam",
          "type": "integer",
        },
      ],
    ],
    "id": "#/simple/test/get",
    "name": "RqstSimpleTestGet",
    "required": Array [],
    "type": "object",
  },
]
`;

exports[`parse parameters Should parse a basic parameter 1`] = `
Array [
  Object {
    "additionalProperties": false,
    "fields": Array [
      Array [
        "param1",
        Object {
          "id": "#/simple/test/get/param1",
          "name": "RqstSimpleTestGet_Field_param1",
          "type": "string",
        },
      ],
    ],
    "id": "#/simple/test/get",
    "name": "RqstSimpleTestGet",
    "required": Array [],
    "type": "object",
  },
]
`;

exports[`parse parameters Should parse global parameters 1`] = `
Array [
  Object {
    "additionalProperties": false,
    "fields": Array [
      Array [
        "globalParam1",
        Object {
          "id": "#/parameters/globalParam1",
          "name": "parameters_Field_globalParam1",
          "type": "string",
        },
      ],
      Array [
        "param1",
        Object {
          "id": "#/simple/test/get/param1",
          "name": "RqstSimpleTestGet_Field_param1",
          "type": "string",
        },
      ],
    ],
    "id": "#/simple/test/get",
    "name": "RqstSimpleTestGet",
    "required": Array [],
    "type": "object",
  },
]
`;

exports[`parse parameters Should parse multiple parameters 1`] = `
Array [
  Object {
    "additionalProperties": false,
    "fields": Array [
      Array [
        "param1",
        Object {
          "id": "#/simple/test/get/param1",
          "name": "RqstSimpleTestGet_Field_param1",
          "type": "string",
        },
      ],
      Array [
        "param2",
        Object {
          "id": "#/simple/test/get/param2",
          "name": "RqstSimpleTestGet_Field_param2",
          "type": "boolean",
        },
      ],
    ],
    "id": "#/simple/test/get",
    "name": "RqstSimpleTestGet",
    "required": Array [],
    "type": "object",
  },
]
`;

exports[`parse parameters Should parse path parameters 1`] = `
Array [
  Object {
    "additionalProperties": false,
    "fields": Array [
      Array [
        "pathParam1",
        Object {
          "id": "#/simple/test/pathParam1",
          "name": "SimpleTest/_Field_pathParam1",
          "type": "string",
        },
      ],
      Array [
        "param1",
        Object {
          "id": "#/simple/test/get/param1",
          "name": "RqstSimpleTestGet_Field_param1",
          "type": "string",
        },
      ],
    ],
    "id": "#/simple/test/get",
    "name": "RqstSimpleTestGet",
    "required": Array [],
    "type": "object",
  },
]
`;

exports[`parse response Should parse a basic response 1`] = `
Array [
  Object {
    "additionalProperties": undefined,
    "fields": Array [
      Array [
        "resp1",
        Object {
          "id": "#/simple/test/get/200/Field_resp1",
          "name": "RespSimpleTestGet200_Field_resp1",
          "type": "string",
        },
      ],
    ],
    "id": "#/simple/test/get/200",
    "name": "RespSimpleTestGet200",
    "required": Array [],
    "type": "object",
  },
  Object {
    "choices": Array [
      Object {
        "id": "/simple/test/choices/RespSimpleTestGet200",
        "name": "RespSimpleTestGet_Ref_RespSimpleTestGet200",
        "ref": "#/simple/test/get/200",
        "type": "ref",
      },
    ],
    "id": "#//simple/test/get",
    "name": "RespSimpleTestGet",
    "required": Array [],
    "type": "anyOf",
  },
]
`;
